system/java_certs.py

```diff
@@ -0,0 +1,171 @@
+options:
+  cert_url:
+    description:
+      - Basic URL to fetch SSL certificate from
+    required: true
+  cert_port:
+    description:
+      - Port to connect to URL. This will be used to create server URL:PORT
+    required: false
+    default: 443
+  cert_path:
+    description:
+      - Local path to load certificate from
+    required: false
+    default: null
+  keystore_path:
+    description:
+      - Path to keystore.
+    required: false
+    default: null
+  keystore_pass:
+    description:
+      - Keystore password
+    required: false
+    default: null
+  state:
+    description:
+      - Defines action which can be either certificate import or removal.
+    choices: [ 'present', 'absent' ]
+    default: 'present'
+    required: false
+
+author: Adam Hamsik
+'''
+
+EXAMPLES = '''
+# Import SSL certificate from google.com to a given cacerts keystore
+java_certs: cert_url=google.com cert_port=443 keystore_path=/usr/lib/jvm/jre7/lib/security/cacerts keystore_pass=changeit state=present
+
+# Remove certificate with given alias from a keystore
+java_certs: cert_url=google.com keystore_path=/usr/lib/jvm/jre7/lib/security/cacerts keystore_pass=changeit state=absent
+'''
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+import os
+import subprocess
+import sys
+import datetime
+import syslog
+
+def import_cert_url(module, url, port, keystore_path, keystore_pass, alias):
+    fetch_cmd = "keytool -printcert -rfc -sslserver %s:%s" % (url, port)
+    import_cmd = "keytool -importcert -noprompt -keystore '%s' -storepass '%s' -alias '%s'" % (keystore_path, keystore_pass, alias)
+
+    # Fetch SSL certificate from remote host.
+    (rc, fetch_out, fetch_err) = module.run_command(fetch_cmd, check_rc=True)
+
+    # Use remote certificate from remote host and import it to a java keystore
+    (rc, import_out, import_err) = module.run_command(import_cmd, data=fetch_out, check_rc=False)
+    if not rc:
+        return module.exit_json(changed=True, msg=import_out,
+            rc=rc, cmd=import_cmd, stdout_lines=import_out)
+    else:
+        if 'already exists' in import_out:
+            return module.exit_json(changed=False, msg=import_out,
+                rc=0, cmd=import_cmd, stdout_lines=import_out)
+        else:
+            return module.fail_json(msg=import_out, rc=rc, cmd=import_cmd)
+
+
+def import_cert_path(module, path, keystore_path, keystore_pass):
+    import_cmd = "keytool -importcert -noprompt -keystore '%s' -storepass '%s' -file '%s'" % (keystore_path, keystore_pass, path)
+
+    # Use local certificate from local path and import it to a java keystore
+    (rc, import_out, import_err) = module.run_command(import_cmd, check_rc=False)
+    if not rc:
+        return module.exit_json(changed=True, msg=import_out,
+            rc=rc, cmd=import_cmd, stdout_lines=import_out)
+    else:
+        if 'already exists' in import_out:
```

Comment:
Is it possible to check if the cert exists in the keystore before trying to import it?